### ISSUE TITLE
FEATURE - Support OpenOCD

### ISSUE THREAD

**Issue by Brando753** [ryankurte/efm32-base#2]
I was curious if OpenOCD can be used instead of the Jlink software in order to use a more open debugging and flashing environment. If so it might be worth adding support in the make files for flashing with openocd commands.

**Comment by ryankurte** [comment]
Hey,

Definitely can be supported, I just haven't had cause to use OpenOCD yet. I do a plan to add support for the open source BlackMagic probe next time I have a compatible project.

If you would like to add it, it just needs needs an equivalent OpenOCD version of the [jlink](https://github.com/ryankurte/efm32-base/blob/master/toolchain/jlink.cmake) configuration file.

**Comment by Brando753** [comment]
Yup, I will look into the code base and open a pull request when I get it supported, just wanted to open a feature request for the work.

**Comment by Brando753** [comment]
So before I started modifying that configuration file I was testing the openocd commands manually, and I could not get them to work with my new EFM32HG dev board. I had used openOCD before with these boards so I grabbed my old Zero Gecko dev board and the commands worked...I am filing an issue with openOCD to see whats going on with the new firmware on these dev boards. I will update here once I get an update from the openOCD project.

**Comment by ryankurte** [comment]
I found a similar issue implementing the [Blackmagic Probe](https://github.com/ryankurte/efm32-base/blob/feature/blackmagic-support/toolchain/bmpm.cmake) which was because the firmware didn't yet support the EFM32PG IC I was using.

**Comment by dholth** [comment]
My attempt to support more geckos with the BMP. https://github.com/blacksphere/blackmagic/pull/264

It would have made more sense to just buy the jlink (as built into the official dev board)

**Comment by ryankurte** [comment]
Yeah I had a go at [simplifying your original PR](https://github.com/ryankurte/blackmagic/tree/feature/efm32-draco-target) too before writing the whole thing off and going back to my jlink :-/

**Comment by ryankurte** [comment]
FYI with simplicity studio you can switch the debugger on any silicon labs dev board to the external 10/20 pin cortex debug connector and use it with other boards. It's not nice and open source, but it sure is convenient.

**Comment by Brando753** [comment]
I will take a look at the OpenOCD implementation when I get a chance later this week will let you know how it goes.

**Comment by dholth** [comment]
@ryankurte do you have a favorite silabs dev board?

**Comment by ryankurte** [comment]
I like the [Pearl Gecko Starter Kit](https://www.silabs.com/products/development-tools/mcu/32-bit/efm32-pearl-gecko-starter-kit) because it's the new architecture / has full pin muxing / the peripherals are bad ass.
I've also got a [Giant Gecko GG1 Starter Kit](https://www.silabs.com/products/development-tools/mcu/32-bit/efm32-giant-gecko-gg11-starter-kit) which is kindof exciting because it's the first new-architecture chip from silabs to have ethernet and USB but I haven't played with it much just yet.
They're both expensive / higher end ICs to actually design with, but they're great for ease of prototyping.
